ambry-router/src/main/java/com/github/ambry/router/BatchOperationCallbackTracker.java

```diff
@@ -78,6 +85,13 @@
    * @param e the {@link Exception} that occurred (if any).
    */
   private void complete(Exception e) {
+    if(quotaChargeCallback != null) {
+      try {
+        quotaChargeCallback.chargeQuota();
+      } catch (RouterException rEx) {
+        LOGGER.info("Exception {} while charging quota for ttl operation", rEx.toString());
+      }
+    }
```

Comment:
we should move this blob inside of the if blob below, like
```
    if (completed.compareAndSet(false, true)) {
        if(quotaChargeCallback != null) {
            try {
                quotaChargeCallback.chargeQuota();
            } catch (RouterException rEx) {
                LOGGER.info("Exception {} while charging quota for ttl operation", rEx.toString());
            }
      }
      NonBlockingRouter.completeOperation(futureResult, callback, null, e, false);
    }
```

Reply:
fixed.